installer/src/main/java/com/teamscale/profiler/installer/Installer.java

```diff
@@ -0,0 +1,192 @@
+package com.teamscale.profiler.installer;
+
+import com.teamscale.profiler.installer.steps.IStep;
+import com.teamscale.profiler.installer.steps.InstallAgentFilesStep;
+import com.teamscale.profiler.installer.steps.InstallEtcEnvironmentStep;
+import com.teamscale.profiler.installer.steps.InstallSystemdStep;
+import org.conqat.lib.commons.collections.CollectionUtils;
+import org.conqat.lib.commons.system.SystemUtils;
+import picocli.CommandLine;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+/** Installs the agent system-globally. */
+public class Installer {
+
+	private static final Path DEFAULT_INSTALL_DIRECTORY = Paths.get("/opt/teamscale-profiler/java");
+	private static final Path DEFAULT_ETC_DIRECTORY = Paths.get("/etc");
+
+	private static final String RERUN_ADVICE = getRerunAdvice();
+
+	private static String getRerunAdvice() {
+		if (SystemUtils.isWindows()) {
```

Comment:
ah yes, I got carried away here and started thinking about how/if this could work on windows :P do you want me to remove it or leave it in for later when we implement Windows support? It's sort of dead code ATM, so can't be triggered (since we don't compile for Windows)

Reply:
You can leave it. Was just confused :D